File: 프로그래머스/lv2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    
    total = brown + yellow
    
    w = 3
    
    while True:
        for h in range(3,w+1):
            
            if(h*w != total):
                continue
            
            if((w-2) * (h-2) == yellow):
                answer.append(w)
                answer.append(h)
                break
        
        if(len(answer) != 0):
            break
        
        w +=1

    return answer

File: 프로그래머스/lv2/42842. 카펫/카펫.swift
import Foundation

func solution(_ brown:Int, _ yellow:Int) -> [Int] {
    
    
    var w:Int = 3
    
    var ans:[Int] = .init()
    
    let total:Int = brown + yellow
    
    while true {
        
        for h in 3...w{
            
            if(h*w != total){
                continue
            }
                
            
            if (w-2) * (h-2) == yellow {
                ans.append(w)
                ans.append(h)
                break
            }
            
        }
        
        if !ans.isEmpty{
            break
        }
        
        w += 1
    }
    
    
    return ans
}